### PR TITLE
[Documentation] Add documentation for class HierarchicalIterativeSolver.

### PR DESCRIPTION
**Pull request by florent-lamiraux**

### CODE REVIEW

File: include/hpp/constraints/iterative-solver.hh [hunk 1]
```diff
@@ -79,6 +83,60 @@ namespace hpp {
...
 
+    /// Solve a system of non-linear equations on a robot configuration
+    ///
+    /// The non-linear system of equations is built by adding equations with
+    /// method HierarchicalIterativeSolver::add.
+    ///
+    /// Note that a hierarchy between the equations can be
+    /// provided. In this case, the solver will try to solve the
+    /// highest priority equations first, and then to solve the lower priority
+    /// equations.
```
Comment (jmirabel) [inline include/hpp/constraints/iterative-solver.hh]: I would mention (here or at HierarchicalIterativeSolver::add):

The priorities are by decreasing order (0 has the priority over 1).

File: include/hpp/constraints/iterative-solver.hh [hunk 2]
```diff
@@ -148,9 +208,31 @@ namespace hpp {
...
+        ///
+        /// \param arg initial guess,
+        /// \param ls line search method used.
+        ///
+        /// Use Newton Rhapson like iterative method until the error is below
+        /// the threshold, or until the maximal number of iterations has been
+        /// reached.
+        ///
+        /// \note Explicit constraints are expressed in their implicit
+        ///       form.
```
Comment (jmirabel) [inline include/hpp/constraints/iterative-solver.hh]: > Explicit constraints are expressed in their implicit form.

What do you mean ?

Comment (florent-lamiraux) [reply include/hpp/constraints/iterative-solver.hh]: I mean that q2 = f (q1) is replaced by q2 - f (q1) = 0.
Isn't it explicit enough ?

**Comment by jmirabel** [comment]
Thanks.

**Pull request merged**